server/base/src/main/java/org/apache/accumulo/server/conf/store/impl/ZooPropStore.java

```diff
@@ -391,4 +399,26 @@ public PropCache getCache() {
   public @Nullable VersionedProperties getWithoutCaching(PropStoreKey<?> propStoreKey) {
     return cache.getWithoutCaching(propStoreKey);
   }
+
+  @Override
+  public boolean validateDataVersion(PropStoreKey<?> storeKey, long expectedVersion) {
+    try {
+      Stat stat = zrw.getStatus(storeKey.getPath());
+      log.trace("data version sync: stat returned: {} for {}", stat, storeKey);
+      if (stat == null || expectedVersion != stat.getVersion()) {
+        propStoreWatcher.signalZkChangeEvent(storeKey);
+        return false;
+      }
+    } catch (InterruptedException ex) {
+      Thread.currentThread().interrupt();
+      throw new IllegalStateException(ex);
+    } catch (KeeperException.NoNodeException ex) {
+      propStoreWatcher.signalZkChangeEvent(storeKey);
```

Comment:
you are returning `true` here?

Reply:
I was allowing the `signalZkChangeEvent` to propagate the change (this removes it from the cache) - I'll change it to return false so that it is immediacy removed from the snapshot too.

Reply:
Fixed in 2ae1730977